interfaces/builtin/microceph_support.go

```diff
@@ -0,0 +1,60 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const microcephSupportSummary = `allows operating as the MicroCeph service`
+
+const microcephSupportBaseDeclarationPlugs = `
+  microceph-support:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
```

Comment:
this needs to define and use also a  microcephSupportBaseDeclarationSlots similar to what microstack-support does